### DISCUSSION TITLE
spec of all fields and what is optional

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#8627] [Q&A] [answered]
maybe biggest challenge i'm having is knowing what fields i can use in my workflow/template/cron yamls

**Comment by terrytangyuan** [comment, marked as answer]
https://argoproj.github.io/argo-workflows/fields/

**Reply by tooptoop4** [reply]
@terrytangyuan 
for a new person it is hard to use and there are several improvements that would help:
1. it does not say which field is mandatory vs optional
2. it does not have a default value column
3. it does not show the range of accepted possible values (ie for 'shutdown' field)
4. it does not show whether multiple levels of values are ok
5. it is not ordered by most common fields
6. it does not have an indication of which field is exclusive with another field or requires another field
7. it does not have grouping of the types, ie many different types (ie GCS, HDFS..etc) of ArtifactRepository in the table of contents could be under ArtifactRepository

**Reply by terrytangyuan** [reply]
These are autogenerated but there's definitely a lot of room for improvements. Feel free to send us PRs to improve the docs!
